src/PostgREST/DbStructure.hs

```diff
@@ -311,22 +313,68 @@ accessibleTables =
       relname as table_name,
       d.description as table_description,
       (
-        c.relkind in ('r', 'v', 'f')
-        and (pg_relation_is_updatable(c.oid::regclass, false) & 8) = 8
-        -- The function `pg_relation_is_updateable` returns a bitmask where 8
-        -- corresponds to `1 << CMD_INSERT` in the PostgreSQL source code, i.e.
-        -- it's possible to insert into the relation.
-        or (exists (
-          select 1
-          from pg_trigger
-          where
-            pg_trigger.tgrelid = c.oid
-            and (pg_trigger.tgtype::integer & 69) = 69)
-            -- The trigger type `tgtype` is a bitmask where 69 corresponds to
-            -- TRIGGER_TYPE_ROW + TRIGGER_TYPE_INSTEAD + TRIGGER_TYPE_INSERT
-            -- in the PostgreSQL source code.
+        (
+          has_table_privilege(c.oid, 'INSERT')
+          or has_any_column_privilege(c.oid, 'INSERT')
+        )
+        and
+        (
+          c.relkind in ('r', 'v', 'f')
+          and (pg_relation_is_updatable(c.oid::regclass, false) & 8) = 8
+          -- The function `pg_relation_is_updateable` returns a bitmask where 8
+          -- corresponds to `1 << CMD_INSERT` in the PostgreSQL source code, i.e.
+          -- it's possible to insert into the relation.
+          or (exists (
+            select 1
+            from pg_trigger
+            where
+              pg_trigger.tgrelid = c.oid
+              and (pg_trigger.tgtype::integer & 69) = 69)
+              -- The trigger type `tgtype` is a bitmask where 69 corresponds to
+              -- TRIGGER_TYPE_ROW + TRIGGER_TYPE_INSTEAD + TRIGGER_TYPE_INSERT
+              -- in the PostgreSQL source code.
+          )
+        )
+      ) as insertable,
+      (
+        (
+          has_table_privilege(c.oid, 'UPDATE')
+          or has_any_column_privilege(c.oid, 'UPDATE')
         )
-      ) as insertable
+        and
+        (
+          c.relkind in ('r', 'v', 'f')
+          and (pg_relation_is_updatable(c.oid::regclass, false) & 20) = 20
+          -- (1 << CMD_UPDATE) + (1 << CMD_DELETE) = 20
```

Comment:
Shouldn't this be 4 or 16 instead of 20? (Just guessing - but not a combo of update and delete, right?)

Reply:
I was using the [information schema](https://github.com/postgres/postgres/blob/master/src/backend/catalog/information_schema.sql) in postgres source code as reference, had the idea it returned both updatable and deletable together, but checking in more detail [the source code](https://doxygen.postgresql.org/rewriteHandler_8c.html#a52a3ea381d9db1b6e917b7280e3e6d71) it may discriminate the UPDATE/INSERT and DELETE events for views. 
In any case, I agree, it should be as you say, I'll fix it.

Reply:
Those constants are defined here:

https://github.com/postgres/postgres/blob/e717a9a18b2e34c9c40e5259ad4d31cd7e420750/src/include/nodes/nodes.h#L682-L693

So, `CMD_UPDATE` seems to be 4 and `CMD_DELETE` 16. Agree?